force_wfmanager/wfmanager_plugin.py

```diff
@@ -27,4 +34,20 @@ def _create_task(self):
         factory_registry = self.application.get_plugin(
             FACTORY_REGISTRY_PLUGIN_ID)
 
-        return WfManagerTask(factory_registry=factory_registry)
+        wf_manager_task = WfManagerTask(factory_registry=factory_registry)
```

Comment:
You already have this code in the WfManagerTask. Why are you copying it here. The wfmanagertask is responsible for opening a given file, you just have to refactor the UI action in two parts: one that responds to the event, the other that accepts the filename and loads the file.